pkg/packagekit/autoupdates.jsx

```diff
@@ -0,0 +1,341 @@
+
+        // check if we have a day of week
+        if (daysOfWeek.indexOf(words[0]) >= 0)
+            this.day = words.shift()
+        else
+            this.day = ""; // daily
+
+        // now there should only be a time left
+        if (words.length == 1 && validTime.test(words[0]))
+            this.time = words[0].replace(/^0+/, "");
+        else
+            this.supported = false;
+    }
+
+    setConfig(enabled, type, day, time) {
+        const timerConf = "/etc/systemd/system/dnf-automatic-install.timer.d/time.conf";
+        var script = "set -e; ";
+
+        if (type !== null) {
+            let value = (type == "security") ? "security" : "default";
+
+            // normally upgrade_type = should already be in the file, so replace that line;
+            // if it's not already present, append it to the file
+            script += "sed -i '/\\bupgrade_type\\b[ \\t]*=/ { h; s/^.*$/upgrade_type = " + value + "/ }; " +
+                      "$ { x; /^$/ { s//upgrade_type = " + value + "/; H }; x }' /etc/dnf/automatic.conf; ";
+
+        }
+
+        // if we enable through Cockpit, make sure that starting the timer doesn't start the .service right away,
+        // due to the packaged default OnBootSec=1h; just set a reasonable initial time which will trigger the code below
+        if (enabled && !this.enabled && !this.time && !this.day)
+            time = "6:00";
+
+        if (time !== null || day !== null) {
+            if (day === "" && time === "") {
+                // restore defaults
+                script += "rm -f " + timerConf + "; ";
+            } else {
+                if (day == null)
+                    day = this.day;
+                if (time == null)
+                    time = this.time;
+                script += "mkdir -p /etc/systemd/system/dnf-automatic-install.timer.d; ";
+                script += "printf '[Timer]\\nOnBootSec=\\nOnCalendar=" + day + " " + time + "\\n' > " + timerConf + "; ";
+                script += "systemctl daemon-reload; ";
+            }
+        }
+
+        if (enabled !== null) {
+            const rebootConf = "/etc/systemd/system/dnf-automatic-install.service.d/autoreboot.conf";
+
+            script += "systemctl " + (enabled ? "enable" : "disable") + " --now dnf-automatic-install.timer; ";
+
+            if (enabled) {
+                // HACK: enable automatic reboots after updating; dnf-automatic does not leave a log file behind for
+                // deciding whether it actually installed anything, so resort to grepping the journal for the last run
+                // (https://bugzilla.redhat.com/show_bug.cgi?id=1491190)
+                script += "mkdir -p /etc/systemd/system/dnf-automatic-install.service.d; ";
+                script += "printf '[Service]\\nExecStartPost=/bin/sh -ec \"" +
+                          "if systemctl status --no-pager --lines=100 dnf-automatic-install.service| grep -q ===========$$; then " +
```

Comment:
Since we want to grep the journal, what about using journalctl?  Also, does this really only look at the last run, or does it look at the last 100 lines?

Reply:
This is exactly why I can't use `journalctl` for this - it doesn't have an option for "just show me the current run". One can sort of fake it with first doing a `systemctl show`, grab the started time from it, and plug that into `journalctl --since`, but overall that's more overhead than just using this. (I asked the systemd guys about that).

Reply:
Oh, `systemctl status` only shows the journal for the current run?  I didn't know that, nice!